File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string s);
int count_words(string words);
int count_sentences(string sentences);


int main(void)
{
    string text = get_string("Text :");
    float L = (float)count_letters(text) / (float)count_words(text) * (float)100;
    float S = (float)count_sentences(text) / (float)count_words(text) * (float)100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if(index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if(index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}
//count leters
int count_letters(string s)
{
    int length = strlen(s);
    int n_letters = 0;
    for (int i=0; i < length; i++)
    {
        if (isalpha(s[i]) != 0)
        {
            n_letters++;
        }
    }
    return n_letters;
}
//count words
int count_words(string words)
{
    int length = strlen(words);
    int n_words = 0;
    for (int i=0; i < length; i++)
    {
        if(isspace(words[i]) != 0)
        {
            n_words++;
        }
    }
    return n_words;
}
int count_sentences(string sentences)
{
    int length = strlen(sentences);
    int n_sentences = 0;
    for (int i=0; i < length; i++)
    {
        if(sentences[i] == 33 || sentences[i] == 63 || sentences[i] == 46)
        {
            n_sentences++;
        }
    }
    return n_sentences;
}

File: sentimental-readability/readability.py
from cs50 import get_string

# intialize required variables
letters = 0
sentences = 0
words = 0

# obtain the paragraph
para = get_string("Text: ")

# iterate over each character and update the score and sentences
for letter in para:
    if letter.isalpha():
        letters += 1
    elif ord(letter) == ord(" "):
        words += 1
    elif ord(letter) == ord(".") or ord(letter) == ord("?") or ord(letter) == ord("!"):
        sentences += 1

words += 1
# calculate the value of L
L = (letters / words) * 100
#calculate S
S = (sentences / words) * 100
# caculate index
index = 0.0588 * L - 0.296 * S - 15.8

# Print out the grade
if(index >= 16):
    print("Grade 16+")
elif (index < 1):
    print("Before Grade 1")
else:
    grade = round(index, 0)
    print(f"Grade {int(grade)}")
